homeassistant/components/rainforest_raven/sensor.py

```diff
@@ -0,0 +1,193 @@
+"""Sensor entity for a Rainforest RAVEn device."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+    StateType,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_MAC,
+    PERCENTAGE,
+    EntityCategory,
+    UnitOfEnergy,
+    UnitOfPower,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import RAVEnDataCoordinator
+
+
+@dataclass
+class RAVEnSensorEntityDescription(SensorEntityDescription):
+    """A class that describes RAVEn sensor entities."""
+
+    message_key: str | None = None
+    attribute_keys: list[str] | None = None
+
+
+SENSORS = (
+    RAVEnSensorEntityDescription(
+        message_key="CurrentSummationDelivered",
+        translation_key="total_energy_delivered",
+        key="summation_delivered",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
```

Comment:
Is `TOTAL_INCREASING` the right state class? Why not `TOTAL`?

Reply:
The same state class is used in the Rainforest Eagle integration: https://github.com/home-assistant/core/blob/63136572a54c9b443a1eea25bc22334c1009b143/homeassistant/components/rainforest_eagle/sensor.py#L34

My impression was that `TOTAL` would be good for a value with a known or static time window, while `TOTAL_INCREASING` would be better for a value which just keeps going up but may be reset to zero at some arbitrary point. I would think that a utility meter better fits the latter.

Reply:
Sure, if the reading resets to 0 at arbitrary points in time, `TOTAL_INCREASING` should be used 👍